### PR TITLE
Add errored result counts and filter parameter details to Policy Checks API docs

### PR DESCRIPTION
**Pull request by jarrodn-au**
### What

Added details and missing fields from the policy outcome API endpoints for policies in an errored state. Added filter details to the policy set outcomes endpoint.

### Why

Aligns the docs with the implementation, documents the new filter parameters.

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [x] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [x] (N/A) Description links to related pull requests or issues, if any.

#### Content
- [x] (N/A) Redirects have been added for moved, renamed, or deleted pages. This requires a separate PR in the [`terraform-website` repository](https://github.com/hashicorp/terraform-website) `redirects.next.js` file.
- [x] API documentation and the API Changelog have been updated. 
- [x] (N/A) Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [x] (N/A) Pages with related content are updated and link to this content when appropriate.
- [x] (N/A) Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [x] (N/A) New pages have metadata (page name and description) at the top.
- [x] (N/A) New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [ ] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [x] (N/A) UI elements (button names, page names, etc.) are bolded.
- [x] The Vercel website preview successfully deployed.

#### Reviews
- [ ] I or someone else reviewed the content for technical accuracy.
- [ ] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

File: website/docs/cloud-docs/api-docs/policy-checks.mdx [hunk 1]
```diff
@@ -357,9 +363,15 @@ This endpoint supports pagination [with standard URL query parameters](/cloud-do
 |----------------------------------- | ----------------------------------------------------------------------- |
 | `page[number]`                     | **Optional.** If omitted, the endpoint returns the first page.          |
 | `page[size]`                       | **Optional.** If omitted, the endpoint returns 20 agent pools per page. |
+| `filter[n][status]`                | **Optional.** If omitted, the endpoint returns all policies regardless of status. Must be one of: passed, failed, errored. |
+| `filter[n][enforcementLevel]`      | **Optional.** Only used if paired with a non-errored status filter. Must be one of: advisory, mandatory. |
```
Comment (laurapacilio) [inline website/docs/cloud-docs/api-docs/policy-checks.mdx:367]: Should these be code font instead? I just want to set them apart from the typical text a bit.

```suggestion
| `filter[n][status]`                | **Optional.** If omitted, the endpoint returns all policies regardless of status. Must be either "passed", "failed", or "errored". |
| `filter[n][enforcementLevel]`      | **Optional.** Only used if paired with a non-errored status filter. Must be either "advisory" or "mandatory." |
```

File: website/docs/cloud-docs/api-docs/policy-checks.mdx [hunk 2]
```diff
@@ -357,9 +363,15 @@ This endpoint supports pagination [with standard URL query parameters](/cloud-do
...
+
+-> **Note**: The `filter[n]` is a zero indexed filter and can combine filtering combinations of statuses and enforcement levels. Policy outcomes with an errored status do not have an enforcement level.
```
Comment (laurapacilio) [inline website/docs/cloud-docs/api-docs/policy-checks.mdx:369]:
```suggestion
-> **Note**: You can use `filter[n]` to combine combinations of statuses and enforcement levels. Policy outcomes with an errored status do not have an enforcement level.
```

**Review by laurapacilio — APPROVED** [review]
Approved with a few suggestions - thank you!

File: website/docs/cloud-docs/api-docs/policy-checks.mdx (hunk 1 above)
Comment (jarrodn-au) [reply website/docs/cloud-docs/api-docs/policy-checks.mdx:367]: I think in quotes as suggested is good, thank you.

**Pull request merged**